abi/src/main/java/org/web3j/abi/datatypes/Array.java

```diff
@@ -12,6 +12,7 @@
     private String type;
     protected final List<T> value;
 
+    @SafeVarargs
```

Comment:
I used the following trick/hack to make this work on Java 1.6: copy/paste/clone the interface from Java 1.8 directly into the Utils module - with that, we should be able to compile the auto-generated classes, and not have different ones between Android and non Android ones.

Reply:
So according to this post https://stackoverflow.com/questions/16366780/safevarargs-and-java-6-interoperability, as long as someone isn't using Spring or something similar that uses reflection at runtime to get the annotations it won't be a problem.

As I'm not an Android developer I can't say how likely this scenario is.
I presume one alternative would be to place it in a separate module with scope *compileOnly* in the ABI dependency, although it would probably need to be tested to ensure it works ok.

Reply:
This other thing we could do is change the RetentionPolicy of the custom SafeVarargs annotation class to SOURCE or CLASS, and this should eliminate any potential runtime problems.

Reply:
@conor10 I'm having some issues building the android branch:
```
$ ./gradlew check
FAILURE: Build failed with an exception.

* What went wrong:
Gradle 4.3 requires Java 7 or later to run. You are currently using Java 6.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org
```

I am using a version 1.6 of Java to compile, but this does seem to be compatible with Gradle. 
 How do you compile the branch?

Reply:
I have the following two properties set in my build.gradle:

```
org.gradle.java.home = /Library/Java/JavaVirtualMachines/jdk1.7.0_40.jdk/Contents/Home
java6Home=/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home
```